src/raw_transaction.rs

```diff
@@ -125,6 +185,15 @@ mod test {
     use std::io::Read;
     use ethereum_types::H256;
 
+    #[test]
+    fn test_eip2718() {
+        //let txn = Transaction::FeeMarketEIP1559(EIP1559Txn::new(...));
+        let txn = EIP1559Txn::new(...);
+        let txn2 = EIP2930Txn::new(...);
+        let txn_3 = LegacyTxn::new(...);
+        let bytes = txn.sign(secret_key);
```

Comment:
@synlestidae This is what I'm thinking for the user-facing API, just two simple methods, new() and sign(). New gives the user the ability to instantiate a new (raw) transaction. Then, they can sign the transaction, which under the hood handles all the encoding and signing, and returns the byte-encoded transaction. What are your thoughts?